protocol/urls/urls_test.go

```diff
@@ -30,3 +30,40 @@ func TestGetLinkPreviewData(t *testing.T) {
 	require.Error(t, err)
 
 }
+
+func TestGetGiphyPreviewData(t *testing.T) {
+	validGiphyLink := "https://giphy.com/gifs/FullMag-robot-boston-dynamics-dance-lcG3qwtTKSNI2i5vst"
+	previewData, err := GetGiphyPreviewData(validGiphyLink)
+	bostonDynamicsEthGifData := LinkPreviewData{
+		Site:         "GIPHY",
+		Title:        "Boston Dynamics Yes GIF by FullMag - Find & Share on GIPHY",
+		ThumbnailURL: "https://media1.giphy.com/media/lcG3qwtTKSNI2i5vst/giphy.gif",
```

Comment:
If they return different subdomains, I would probably the path and just match against the path, similarly as we do in the `switch/case` statement, not as effective, but should do.